gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -238,11 +241,37 @@ else if (statement instanceof ExecuteImmediate executeImmediate) {
         }
     }
 
-    private Map<String, String> getPreparedStatements(HttpServletRequest request)
+    private void processRequestBody(ContainerRequestContext requestContext)
+    {
+        InputStream entityStream = requestContext.getEntityStream();
+        if (entityStream == null ||
+                !requestContext.hasEntity() || requestContext.getMediaType() == null ||
+                !requestContext.getMediaType().getParameters().containsKey("charset") ||
+                !StandardCharsets.UTF_8.name().equalsIgnoreCase(requestContext.getMediaType().getParameters().get("charset"))) {
+            return;
+        }
+        InputStreamReader entityReader = new InputStreamReader(entityStream, StandardCharsets.UTF_8);
+        try (BufferedReader reader = new BufferedReader(entityReader)) {
+            processRequestBody(reader, getPreparedStatements(requestContext));
+        }
```

Comment:
We should close InputStreamReader explicitly or use try-with-resources for both `InputStreamReader` and `BufferedReader`.